### DISCUSSION TITLE
Mustache Module

### DISCUSSION THREAD

**Discussion by beleon** [janet-lang/janet#1126] [Show and tell]
I made a simple mustache module which uses mustach and cJSON, have a look: [mustache-janet](https://github.com/beleon/mustache-janet)

**Comment by sogaiu** [comment]
Nice -- perhaps this is more complete than [musty](https://github.com/pyrmont/musty).

**Reply by beleon** [reply]
Thanks :) it should support everything that mustach supports, which should be everything, but I haven't tested it extensively yet. However, currently the json input has to be a string, musty supports native janet datastructures (tables, etc.).

**Reply by sogaiu** [reply]
Thanks for the explanation.

I wonder if [this](https://janet-lang.org/api/spork/json.html) could be useful with respect to converting between a string of JSON and a Janet value.

**Reply by beleon** [reply]
That would definitively work, but it's a bit hacky and requires the spork module. The janet datastructures would be encoded to a json string via spork/json/encode and then parsed again by cJSON. I would be cleaner to make mustach work with the janet datastructures directly and maybe I'll implement this at some point. For now I think it's fine to pass the json string directly or to manually use spork to encode janet datastructures into a json string and pass that to mustache/render
